aws/resource_aws_launch_template.go

```diff
@@ -563,10 +571,32 @@ func resourceAwsLaunchTemplateUpdate(d *schema.ResourceData, meta interface{}) e
 			LaunchTemplateData: launchTemplateData,
 		}
 
-		_, createErr := conn.CreateLaunchTemplateVersion(launchTemplateVersionOpts)
+		createVersion, createErr := conn.CreateLaunchTemplateVersion(launchTemplateVersionOpts)
 		if createErr != nil {
 			return createErr
 		}
+
+		if d.HasChange("default_version") {
+			modifyLaunchTemplateOpts := &ec2.ModifyLaunchTemplateInput{
+				LaunchTemplateId: aws.String(d.Id()),
+				DefaultVersion:   aws.String(strconv.Itoa(d.Get("default_version").(int))),
+			}
+			_, modifyErr := conn.ModifyLaunchTemplate(modifyLaunchTemplateOpts)
+			if modifyErr != nil {
+				return modifyErr
+			}
+		}
+
+		if v, vOk := d.GetOk("update_default_version"); vOk && v.(bool) {
```

Comment:
I believe if you update only `update_default_version` in your Terraform configuration (even `true` to `false`), the logic above will always call `CreateLaunchTemplateVersion`. We should prevent this behavior and verify this doesn't occur.